### ISSUE TITLE
what is the output of mx.contrib.sym.ctc_loss() ???

### ISSUE THREAD

**Issue by kobenaxie** [apache/incubator-mxnet#6871]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info 
Operating System:centos 

Compiler:

Package used (Python/R/Scala/Julia):python

MXNet version:0.10

Or if installed from source:pip

MXNet commit hash (`git rev-parse HEAD`):

If you are using python package, please provide

Python version and distribution: python 2.7

## Error Message:
===================================================================

**I want to know what does the mx.contrib.sym.ctc_loss(data, label) return ?**

Signature: mx.contrib.sym.ctc_loss(*args, **kwargs)
Docstring:
Connectionist Temporal Classification Loss.

The shapes of the inputs and outputs:

- **data**: *(sequence_length, batch_size, alphabet_size + 1)*
- **label**: *(batch_size, label_sequence_length)*
- **out**: *(batch_size)*.

``label`` is a tensor of integers between 1 and *alphabet_size*. If a 
sequence of labels is shorter than *label_sequence_length*, use the special 
padding character 0 at the end of the sequence to conform it to the correct 
length. For example, if *label_sequence_length* = 4, and one has two sequences 
of labels [2, 1] and [3, 2, 2], the resulting ```label``` tensor should be 
padded to be::

  [[2, 1, 0, 0], [3, 2, 2, 0]]

The ``data`` tensor consists of sequences of activation vectors. The layer 
applies a softmax to each vector, which then becomes a vector of probabilities 
over the alphabet. Note that the 0th element of this vector is reserved for the 
special blank character.

See *Connectionist Temporal Classification: Labelling Unsegmented 
Sequence Data with Recurrent Neural Networks*, A. Graves *et al*. for more 
information.

Defined in src/operator/contrib/ctc_loss.cc:L79

Parameters
----------
data : Symbol
    Input data to the ctc_loss op.
label : Symbol
    Ground-truth labels for the loss.

name : string, optional.
    Name of the resulting symbol.

Returns
-------
Symbol
    The result symbol.

===================================================================
my network defined below:

def sym_ocrCTC(batch_size=100, num_layers=2, num_hidden=128, seq_len=30):
    stack = mx.rnn.SequentialRNNCell()
    for i in range(num_layers):
        if i==(num_layers-1): # laster layer
            stack.add(mx.rnn.LSTMCell(num_hidden=63, prefix='lstm_l%d_'%i))
        else:
            stack.add(mx.rnn.LSTMCell(num_hidden=num_hidden, prefix='lstm_l%d_'%i))

    data = mx.sym.Variable('data') # has the shape of (Batch_size, 30, 100)
    label = mx.sym.Variable('label')

    stack.reset()
    outputs, states = stack.unroll(seq_len, inputs=data, merge_outputs=True)
    
    pred = mx.sym.swapaxes(data=outputs, dim1=0, dim2=1)
    loss = mx.contrib.sym.ctc_loss(data=pred, label=label)
    ctc_loss = mx.sym.make_loss(loss)
    return ctc_loss

Is there any problem ? @piiswrong

**Comment by piiswrong** [comment]
@sbodenstein

**Comment by sbodenstein** [comment]
> I want to know what does the mx.contrib.sym.ctc_loss(data, label) return ?

It returns the value of the CTC loss for every example in batch of inputs.

I've added a comment about what the output is in the docs: https://github.com/dmlc/mxnet/pull/6906

**Comment by kobenaxie** [comment]
Thank you very much ~ @piiswrong @sbodenstein

**Comment by kobenaxie** [comment]
I reopen this issue, because I have a idea ,"How about add ‘layout=NTC’ parameter in mx.contrib.sym.ctc_loss(data,label,layout) , like unroll() in mxnet.rnn.BaseRNNCell()~", and another question,normally, the output of rnn.unroll() has the shape of (seq_len, batch_size, num_hidden)[layout='NTC'], and the 'data' of mx.contrib.sym.ctc_loss(data,label) need the shape
 (seq_len, batch_size, alphabet_size + 1), it means I need to add a FC layer (with num_hidden=alphabet_size+1) before mx.contrib.sym.ctc_loss ?
@sbodenstein

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
